### PR TITLE
DB-8190 DB-9604 Make Spark fetch and shuffle partitions configurable

### PR DESCRIPTION
**Pull request by dgomezferro**

### CODE REVIEW

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

File: hbase_sql/src/main/java/com/splicemachine/derby/impl/SpliceSpark.java [hunk 1]
```diff
@@ -70,14 +71,23 @@ private SpliceSpark() {} // private constructor forbids creating instances
...
     public static synchronized SparkSession getSession() {
         String threadName = Thread.currentThread().getName();
         if (!SpliceClient.isClient() && !threadName.startsWith("olap-worker-")) {
              // Not running on the Olap Server... raise exception. Use getSessionUnsafe() if you know what you are doing.
             throw new RuntimeException("Trying to get a SparkSession from outside the OlapServer");
         }
-        return getSessionUnsafe();
+        SparkSession result = sessions.get();
+        if (result == null) {
+            result = session.newSession();
```
Comment (jyuanca) [inline hbase_sql/src/main/java/com/splicemachine/derby/impl/SpliceSpark.java:87]: Is it possible session is not initialized and null when getSession s called first time?

File: hbase_sql/src/main/java/com/splicemachine/stream/StreamableRDD.java [hunk 2]
```diff
@@ -153,6 +156,8 @@ private void submitBatch(int batch, int batchSize, int numPartitions, final Java
             public Object call() throws Exception {
                 SparkContext sc = SpliceSpark.getContextUnsafe().sc();
                 sc.setLocalProperties(properties);
+                if (shufflePartitions != null)
+                    sc.setLocalProperty(SQLConf.SHUFFLE_PARTITIONS().key(), shufflePartitions.toString());
```
Comment (yxia92) [inline hbase_sql/src/main/java/com/splicemachine/stream/StreamableRDD.java:160]: Why do we need to set shuffle partition property both here and in QueryJob?

Comment (dgomezferro) [reply hbase_sql/src/main/java/com/splicemachine/stream/StreamableRDD.java]: This wasn't needed, it was an initial attempt at fixing the problem that didn't work out, nice catch!

File: hbase_sql/src/main/java/com/splicemachine/derby/impl/SpliceSpark.java (hunk 1 above)
Comment (dgomezferro) [reply hbase_sql/src/main/java/com/splicemachine/derby/impl/SpliceSpark.java]: Nice catch, I'm calling getSessionUnsafe() now to guarantee it's properly initialized

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED

**Pull request merged**